scripts/docs_link_checker.py

```diff
@@ -0,0 +1,306 @@
+        logger.debug("Cheking absolute image %s in file %s", link, file)
+
+        image_file = self._get_absolute_path(path)
+        if not os.path.isfile(image_file):
+            logger.info("Absolute image %s in file %s was not found", link, file)
+            return LinkReport(link, file, f"Image {image_file} not found")
+        else:
+            logger.debug("Absolute image %s in file %s found", link, file)
+            return None
+
+    def _check_relative_link(self, link: str, file: str, path: str) -> LinkReport:
+        logger.debug("Checking relative link %s in file %s", link, file)
+
+        md_file = self._get_relative_path(file, path)
+        logger.debug("Relative link %s resolved to path %s", link, md_file)
+
+        if not os.path.isfile(md_file):
+            logger.info("Relative link %s in file %s was not found", link, file)
+            return LinkReport(link, file, f"Linked file {md_file} not found")
+        else:
+            logger.debug("Relative link %s in file %s found", link, file)
+            return None
+
+    def _check_relative_image(self, link: str, file: str, path: str) -> LinkReport:
+        logger.debug("Cheking relative image %s in file %s", link, file)
+
+        image_file = self._get_relative_path(file, path)
+        if not os.path.isfile(image_file):
+            logger.info("Relative image %s in file %s was not found", link, file)
+            return LinkReport(link, file, f"Image {image_file} not found")
+        else:
+            logger.debug("Relative image %s in file %s found", link, file)
+            return None
+
+    def _check_docroot_link(self, link: str, file: str, path: str) -> LinkReport:
+        logger.debug("Checking docroot link %s in file %s", link, file)
+
+        md_file = self._get_docroot_path(path)
+        if not os.path.isfile(md_file):
+            logger.info("Docroot link %s in file %s was not found", link, file)
+            return LinkReport(link, file, f"Linked file {md_file} not found")
+        else:
+            logger.debug("Docroot link %s in file %s found", link, file)
+            return None
+
+    def check_link(self, match: re.Match, file: str) -> LinkReport:
+        """Checks that a link is valid. Valid links are:
+        - Absolute links that begin with a forward slash and the specified site prefix (ex: /docs) with no suffix
+        - Absolute images with an image suffix
+        - Relative links that begin with either . or .. and have a .md suffix
+        - Relative images with an image suffix
+        - Docroot links that begin with a character (neither . or /) are relative to the doc root (ex: /docs) and have a .md suffix
+
+        Args:
+            match: A positive match of a markdown link (ex: [...](...))
+            file: The file where the match was found
+
+        Returns:
+            A LinkReport if the link is broken, otherwise None
+        """
+        link = match.group(2)
```

Comment:
Are we confident this will always grab a value? Should we check the size of the matches before asking for item 2 just in case?

Reply:
There will always be 2, otherwise the match would fail